clang/lib/CIR/CodeGen/CIRGenOpenMPRuntime.cpp

```diff
@@ -52,3 +52,13 @@ bool CIRGenOpenMPRuntime::emitTargetGlobal(clang::GlobalDecl &GD) {
   assert(!UnimplementedFeature::openMPRuntime());
   return false;
 }
+
+void CIRGenOpenMPRuntime::emitTaskWaitCall(CIRGenFunction &CGF,
+                                           mlir::Location Loc,
+                                           const OMPTaskDataTy &Data,
+                                           mlir::OpBuilder &builder) {
+  if (!CGF.HaveInsertPoint())
+    return;
+  // This could change in the near future when OpenMP 5.0 gets supported by MLIR
+  builder.create<mlir::omp::TaskwaitOp>(Loc);
```

Comment:
Thanks for clarifying this to me! I had some confusion related to OpenMPIRBuilder, I thought we are always using it indirectly through MLIR, but maybe we would want to support OpenMP through LLVM IR like in the OG clang? If that is the case know I understand and I will do something similar for barrier, but if that is not the case, then a lot of code can be reduced

Reply:
The golden rule is to follow the original codegen the closest as possible, because that's the status quo on how OpenMP gets handled in clang. Of course, you should only fill in the paths you care about, but we should maintain the skeleton in case other parties care about other paths.